src/Nat/add/succ_right.fm

```diff
@@ -0,0 +1,12 @@
+Nat.add.succ_right(a: Nat, b: Nat)
+  : (a + Nat.succ(b)) == Nat.succ(a + b)
+  case a {
+    zero: refl
+    succ:
+      let h = Nat.add.succ_right(a.pred, b)
+      let h1 = Equal.apply!!<Nat.add(a.pred, Nat.succ(b))
```

Comment:
I'm avoiding big line idents like that, instead I'd move everything to the line below, like:

```
let h1 = Equal.apply!!<
  Nat.add(a.pred, Nat.succ(b)),
  Nat.succ(Nat.add(a.pred,b)),
  Nat.succ>
 ```

Or give names like:

```
def e0 = Nat.add(a.pred, Nat.succ(b))
def e1 =  Nat.succ(Nat.add(a.pred,b))
let h1 = Equal.apply!!<e0, e1, Nat.succ>
```